tests/schema.py

```
@@ -758,3 +780,26 @@ def test_truncate_table(self):
+            self.database.pragma('foreign_keys', 'on')
+
+    def test_named_constraints_integration(self):
+        t = TMNamedConstraints.create(k='k1', v=1)  # Sanity test.
+        fails = [
+            {'fk': t.id - 1, 'k': 'k2', 'v': 1},  # Invalid fk.
+            {'fk': t.id, 'k': 'k3', 'v': 0},  # Invalid val.
+            {'fk': t.id, 'k': 'kx', 'v': 1}]  # Invalid key.
+        for f in fails:
+            # MySQL may use OperationalError.
+            with self.assertRaises((IntegrityError, OperationalError)):
+                with self.database.atomic() as tx:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)